src/sentry/ingest/consumer/simple_event.py

```diff
@@ -36,19 +46,45 @@ def process_simple_event_message(
         tags={"consumer": consumer_type},
         unit="byte",
     )
-    message: IngestMessage = msgpack.unpackb(raw_payload, use_list=False)
 
-    message_type = message["type"]
-    project_id = message["project_id"]
+    try:
+        message: IngestMessage = msgpack.unpackb(raw_payload, use_list=False)
+
+        message_type = message["type"]
+        project_id = message["project_id"]
 
-    if message_type != "event":
-        raise ValueError(f"Unsupported message type: {message_type}")
+        if message_type != "event":
+            raise ValueError(f"Unsupported message type: {message_type}")
 
-    try:
-        with metrics.timer("ingest_consumer.fetch_project"):
-            project = Project.objects.get_from_cache(id=project_id)
-    except Project.DoesNotExist:
-        logger.exception("Project for ingested event does not exist: %s", project_id)
-        return
+        try:
+            with metrics.timer("ingest_consumer.fetch_project"):
+                project = Project.objects.get_from_cache(id=project_id)
+        except Project.DoesNotExist:
+            logger.exception("Project for ingested event does not exist: %s", project_id)
+            return
+
+        return process_event(message, project, reprocess_only_stuck_events)
+
+    except Exception:
+        # Any exception that fails schema validation will raise InvalidMessage, which Arroyo will DLQ.
+        # Messages that pass schema validation will not be DLQed as they may be retriable.
+
+        default_topic = consumer_type_to_default_topic[consumer_type].value
+
+        # TODO: Currently, there is only a schema for ingest-events, so just continue to re-raise
+        # the exception if it's a different topic. This can be removed once attachments and transactions
+        # have schemas too.
+        if default_topic != "ingest-events":
+            raise
+
+        codec = sentry_kafka_schemas.get_codec(default_topic)
```

Comment:
Isn't this a heavy operation ? Any reason not to instantiate the codec only once ?

Reply:
it's cached in the library